packages/presentation/lib/src/pages/login_page/bloc/login_bloc.dart

```diff
@@ -40,27 +51,38 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
   final LoginGoogleUseCase loginGoogleUseCase;
   final LoginFaceBookUseCase loginFaceBookUseCase;
   final LogButtonUseCase logButton;
-  final LoginValidator loginValidator;
+  final ValidateLoginFormUseCase formValidator;
+  final GlobalKey<FormState> formKey;
+  final ResultToLocalizedMapper localizationResultMapper;
+  ValidationResult? validationResult = null;
 
   _LoginBloc(
     this.loginWithEmailAndPass,
     this.loginGoogleUseCase,
     this.loginFaceBookUseCase,
     this.logButton,
-    this.loginValidator,
+    this.formValidator,
+    this.formKey,
+    this.localizationResultMapper,
   ) : super(LoginData.init());
 
+  @override
+  void init() {
+    emit(data: tile);
+    super.init();
+  }
+
   @override
   Future<void> auth() async {
-    emit(data: tile, isLoading: false);
-    if (!loginValidator(tile.login, tile.password)) {
-      emit(data: tile.copyWith(errorMessage: 'Fill in your login or password'));
-      return;
-    }
-    emit(data: tile, isLoading: true);
     logButton(EventName.emailAndPasswordBtn);
     final UserEmailPass user = UserEmailPass(tile.login, tile.password);
-    _tryLogin(await loginWithEmailAndPass(user));
+    try {
+      await formValidator(user);
+      _tryLogin(await loginWithEmailAndPass(user));
```

Comment:
need to be refactored a bit more. u can avoid passing bool into `_tryLogin`. if u will throw an exception in case auth failed, u can have the following flow:

try {
await validator()
await login()
navigateToLoggedInScreen()
} on ValidationException catch